internal/luks2/cryptsetup.go

```diff
@@ -363,67 +344,40 @@ func AddKey(devicePath string, existingKey, key []byte, options *AddKeyOptions)
 		options = &AddKeyOptions{Slot: AnySlot}
 	}
 
-	fifoPath, cleanupFifo, err := mkFifo()
-	if err != nil {
-		return xerrors.Errorf("cannot create FIFO for passing existing key to cryptsetup: %w", err)
-	}
-	defer cleanupFifo()
-
 	args := []string{
 		// add a new key
 		"luksAddKey",
 		// LUKS2 only
 		"--type", "luks2",
-		// read existing key from named pipe
-		"--key-file", fifoPath}
+		// read existing key from stdin, specifying key size so
+		// cryptsetup knows where the existing key ends and the new key
+		// starts (we are passing both keys via stdin). Otherwise it
+		// would interpret new lines as separator for the keys, while
+		// we actually allow '\n' to be part of the keys.
+		"--key-file", "-",
+		"--keyfile-size", strconv.Itoa(len(existingKey)),
+		// remove warnings and confirmation questions
+		"--batch-mode"}
```

Comment:
Just a minor nit, but this file also uses the short form `-q` option in `Format` - they should probably be consistent.

Reply:
Good point, I've changed to the long option in `Format`